### PR TITLE
Add providesGVR to patch file and remove from package.yaml

### PR DESCRIPTION
**Pull request by brendarearden**
Add providesGVR to patch file and remove from package.yaml

### CODE REVIEW

**Comment by aiyengar2** [comment]
Related PR: https://github.com/rancher/charts/pull/620, which deprecates this flag via the package.yaml

**Comment by brendarearden** [comment]
Monitoring and Kiali patch files had the gvr added to them with this pr https://github.com/rancher/charts/pull/622

**Review by aiyengar2 — APPROVED** [review]
LGTM, thanks for the quick PR!

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 1]
```diff
@@ -1,7 +1,7 @@
 diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/Chart.yaml packages/rancher-monitoring/charts/Chart.yaml
 --- packages/rancher-monitoring/charts-original/Chart.yaml
 +++ packages/rancher-monitoring/charts/Chart.yaml
-@@ -1,22 +1,29 @@
+@@ -1,22 +1,28 @@
```
Comment (cbron) [inline packages/rancher-monitoring/rancher-monitoring.patch:4]: issue here ^ no changes in file but lopped off a line

**Review by cbron — CHANGES_REQUESTED** [review]
need to delete tgz's

File: packages/rancher-kiali-server/rancher-kiali-server.patch [hunk 2]
```diff
@@ -1,7 +1,7 @@
 diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-kiali-server/charts-original/Chart.yaml packages/rancher-kiali-server/charts/Chart.yaml
 --- packages/rancher-kiali-server/charts-original/Chart.yaml
 +++ packages/rancher-kiali-server/charts/Chart.yaml
-@@ -1,20 +1,28 @@
+@@ -1,20 +1,26 @@
```
Comment (aiyengar2) [inline packages/rancher-kiali-server/rancher-kiali-server.patch:4]: Do we need this change as well or should this be removed like `rancher-monitoring`?

Comment (brendarearden) [reply packages/rancher-kiali-server/rancher-kiali-server.patch]: good catch! I removed this change as well.

**Review by aiyengar2 — APPROVED** [review]

File: packages/rancher-gatekeeper/rancher-gatekeeper.patch [hunk 3]
```diff
@@ -19,6 +19,8 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-gatekeeper/charts-original/Cha
 +  catalog.cattle.io/experimental: true
 +  catalog.cattle.io/namespace: cattle-gatekeeper-system
 +  catalog.cattle.io/release-name: rancher-gatekeeper
++  catalog.cattle.io/auto-install: rancher-gatekeeper-crd=match
++  catalog.cattle.io/provides-gvr: configs.config.gatekeeper.sh/v1alpha1
```
Comment (cbron) [inline packages/rancher-gatekeeper/rancher-gatekeeper.patch:23]: Lots of incorrect gvrs in here but I will follow up to that later, its unrelated.

**Review by cbron — APPROVED** [review]
lgtm

**Pull request merged**